tests/dualtor/test_tunnel_memory_leak.py

```diff
@@ -0,0 +1,183 @@
+"""
+1. On a dual ToR testbed, confirm that the tunnel packet handler service is running 
+in the SWSS container on active Tor (supervisorctl status tunnel_packet_handler)
+2. Send a continuous stream of IPinIP packets similar to those sent from the standby
+ToR to the active ToR
+3. Check SWSS container memory consumption
+"""
+import pytest
+import logging
+import random
+import re
+import contextlib
+from threading import Thread
+from ptf import testutils
+from tests.common.dualtor.mux_simulator_control import toggle_all_simulator_ports_to_upper_tor
+from tests.common.dualtor.dual_tor_utils import upper_tor_host, lower_tor_host
+from tests.common.dualtor.server_traffic_utils import ServerTrafficMonitor
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.dualtor.dual_tor_utils import get_t1_ptf_ports
+from tests.common.dualtor.dual_tor_utils import mux_cable_server_ip
+from tests.common.dualtor.dual_tor_utils import build_packet_to_server
+from tests.common.dualtor.dual_tor_utils import flush_neighbor
+from tests.common.helpers.dut_utils import get_program_info
+
+
+pytestmark = [
+    pytest.mark.topology("dualtor")
+]
+
+PACKET_COUNT = 1000
+swss_mem_percent = 0
+
+
+def validate_neighbor_entry_exist(duthost, neighbor_addr):
+    """Validate if neighbor entry exist on duthost
+
+    Args:
+        duthost (AnsibleHost): Device Under Test (DUT)
+        neighbor_addr (str): neighbor's ip address 
+
+    Returns:
+        bool: True if neighbor exists. Otherwise, return False.
+    """
+    command = "ip neighbor show %s" % neighbor_addr
+    output = [_.strip() for _ in duthost.shell(command)["stdout_lines"]]
```

Comment:
For this specific case, we expect the neighbor entry to be 'FAILED', please add a check for it.

Reply:
@theasianpianist As I mentioned in my case description, I test the whole process, the server will receive the sent packets, `arp_responder `and `icmp_responder `will be started, so neighbor entry will be added with correct one, not "FAILED" entry.
Otherwise, "FAILED" entry will trigger swss to create another invalid tunnel to forward the packet and they will be dropped.

Reply:
Can we check that the entry is 'REACHABLE' in this case

Reply:
Done